commit 8e02d147299c68146eb4eba5bbe34fc3d3361866
Author: Aleix Pol <aleixpol@kde.org>
Date:   Tue Jul 21 19:09:15 2026 +0200

    flatpak-builder-to-bst: Address globbed parent directories

diff --git a/utils/flatpak-builder-to-bst.py b/utils/flatpak-builder-to-bst.py
index 47dc57ce8..92bc1d5f7 100755
--- a/utils/flatpak-builder-to-bst.py
+++ b/utils/flatpak-builder-to-bst.py
@@ -57,7 +57,9 @@ def relativise_path(path):
 
 def cleanup_platform_split_rules(pattern):
     base = "%{prefix}" + pattern if pattern.startswith("/") else "**/" + pattern
-    if any(char in pattern for char in "*?[]"):
+    # Detect paths to cleanup also when the * is in a parent directory
+    basename = pattern.rsplit("/", maxsplit=1)[-1]
+    if any(char in basename for char in "*?[]"):
         return [base]
     return [base, f"{base}/**"]
 
