diff --git a/elements/components/extra-cmake-modules.bst b/elements/components/extra-cmake-modules.bst
index 1fdbdd77fcc9326d5304368606afff85159f375b..8e0090af6c65b624de486e5e773ad1e0b1744390 100644
--- a/elements/components/extra-cmake-modules.bst
+++ b/elements/components/extra-cmake-modules.bst
@@ -19,4 +19,4 @@ sources:
   track: v*
   exclude:
   - '*rc*'
-  ref: v6.23.0-0-gb70177ddb8387484a1e12ab33f6f15979f730074
+  ref: v6.24.0-0-g862b5b0ed8b2413a6c0e524a13640495dc5bb0d3
