diff -Naur a/cmake/FindWrapProtoc.cmake b/cmake/FindWrapProtoc.cmake
--- a/cmake/FindWrapProtoc.cmake	2023-11-20 12:56:30.000000000 +0100
+++ b/cmake/FindWrapProtoc.cmake	2024-02-01 11:02:36.989111511 +0100
@@ -58,7 +58,7 @@
     endif()
     find_program(__WrapProtoc_protoc_imported_location
         NAMES protoc protoc.exe
-        PATHS "${__WrapProtoc_extra_prefix_paths}" "$ENV{Protobuf_ROOT}"
+        PATHS "${__WrapProtoc_extra_prefix_paths}" "$ENV{Protobuf_ROOT}" "/app"
         PATH_SUFFIXES bin
     )
 endif()
