diff --git a/cmake/FindWrapProtoc.cmake b/cmake/FindWrapProtoc.cmake
index 0ee518b..13d0759 100644
--- a/cmake/FindWrapProtoc.cmake
+++ b/cmake/FindWrapProtoc.cmake
@@ -66,7 +66,7 @@ if(NOT __WrapProtoc_protoc_imported_location)
     endif()
     find_program(__WrapProtoc_protoc_imported_location
         NAMES protoc protoc.exe
-        HINTS "${__WrapProtoc_extra_prefix_paths}" "$ENV{Protobuf_ROOT}"
+        HINTS "${__WrapProtoc_extra_prefix_paths}" "$ENV{Protobuf_ROOT}" /app
         PATH_SUFFIXES bin
         ${__WrapProtoc_find_program_args}
     )
