diff --git a/src/client/qwaylandscreen.cpp b/src/client/qwaylandscreen.cpp
index 5b43428d..680ebb71 100644
--- a/src/client/qwaylandscreen.cpp
+++ b/src/client/qwaylandscreen.cpp
@@ -139,7 +139,8 @@ QList<QPlatformScreen *> QWaylandScreen::virtualSiblings() const
     const QList<QWaylandScreen*> screens = mWaylandDisplay->screens();
     list.reserve(screens.count());
     foreach (QWaylandScreen *screen, screens)
-        list << screen;
+        if (screen->screen())
+		list << screen;
     return list;
 }
 
