diff --git a/src/client/qwaylanddatadevice.cpp b/src/client/qwaylanddatadevice.cpp
index 300c9de..11984f9 100644
--- a/src/client/qwaylanddatadevice.cpp
+++ b/src/client/qwaylanddatadevice.cpp
@@ -111,7 +111,10 @@ void QWaylandDataDevice::startDrag(QMimeData *mimeData, QWaylandWindow *icon)
     if (!origin)
         origin = m_display->currentInputDevice()->touchFocus();
 
-    start_drag(m_dragSource->object(), origin->object(), icon->object(), m_display->currentInputDevice()->serial());
+    if (origin)
+        start_drag(m_dragSource->object(), origin->object(), icon->object(), m_display->currentInputDevice()->serial());
+    else
+        qCDebug(lcQpaWayland) << "Couldn't start a drag because the origin window could not be found.";
 }
 
 void QWaylandDataDevice::cancelDrag()
