diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5181b17..c32f180 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -29,7 +29,7 @@ include(ECMSetupVersion)
 include(ECMGenerateHeaders)
 include(ECMMarkNonGuiExecutable)
 
-if (CMAKE_SYSTEM_NAME MATCHES Linux)
+if (FALSE)
     find_package( UDev )
 
     set_package_properties(UDev PROPERTIES  URL "http://www.kernel.org/pub/linux/utils/kernel/hotplug/udev.html"

