diff --git a/src/kstatusnotifieritemdbus_p.cpp b/src/kstatusnotifieritemdbus_p.cpp
index 943cf7c..9f6d1c4 100644
--- a/src/kstatusnotifieritemdbus_p.cpp
+++ b/src/kstatusnotifieritemdbus_p.cpp
@@ -149,22 +149,22 @@ int KStatusNotifierItemDBus::s_serviceCount = 0;
 KStatusNotifierItemDBus::KStatusNotifierItemDBus(KStatusNotifierItem *parent)
     : QObject(parent),
       m_statusNotifierItem(parent),
-      m_service(QStringLiteral("org.kde.StatusNotifierItem-%1-%2")
+      m_connId(QStringLiteral("org.kde.StatusNotifierItem-%1-%2")
                 .arg(QCoreApplication::applicationPid())
                 .arg(++s_serviceCount)),
-      m_dbus(QDBusConnection::connectToBus(QDBusConnection::SessionBus, m_service))
+      m_dbus(QDBusConnection(m_connId))
 {
+    m_dbus = QDBusConnection::connectToBus(QDBusConnection::SessionBus, m_connId);
+
     new StatusNotifierItemAdaptor(this);
-    qCDebug(LOG_KNOTIFICATIONS) << "service is" << m_service;
-    m_dbus.registerService(m_service);
+    qCDebug(LOG_KNOTIFICATIONS) << "service is" << m_connId;
     m_dbus.registerObject(QStringLiteral("/StatusNotifierItem"), this);
 }

 KStatusNotifierItemDBus::~KStatusNotifierItemDBus()
 {
     m_dbus.unregisterObject(QStringLiteral("/StatusNotifierItem"));
-    m_dbus.unregisterService(m_service);
-    m_dbus.disconnectFromBus(m_service);
+    m_dbus.disconnectFromBus(m_connId);
 }

 QDBusConnection KStatusNotifierItemDBus::dbusConnection() const
@@ -174,7 +174,7 @@ QDBusConnection KStatusNotifierItemDBus::dbusConnection() const

 QString KStatusNotifierItemDBus::service() const
 {
-    return m_service;
+    return m_dbus.baseService();
 }

 bool KStatusNotifierItemDBus::ItemIsMenu() const
diff --git a/src/kstatusnotifieritemdbus_p.h b/src/kstatusnotifieritemdbus_p.h
index 282f0b2..1ea2cd2 100644
--- a/src/kstatusnotifieritemdbus_p.h
+++ b/src/kstatusnotifieritemdbus_p.h
@@ -223,7 +223,7 @@ Q_SIGNALS:

 private:
     KStatusNotifierItem *m_statusNotifierItem;
-    QString m_service;
+    QString m_connId;
     QDBusConnection m_dbus;
     static int s_serviceCount;
 };
