diff --git a/src/runtime/kwallet-query/CMakeLists.txt b/src/runtime/kwallet-query/CMakeLists.txt
index 42b4687..9e02cde 100644
--- a/src/runtime/kwallet-query/CMakeLists.txt
+++ b/src/runtime/kwallet-query/CMakeLists.txt
@@ -1,19 +1,7 @@
-PROJECT(kwallet-dmenu)
+project(kwallet-dmenu)
 
-set(CMAKE_EXPORT_COMPILE_COMMANDS "ON")
-
-FIND_PACKAGE(Qt5 ${REQUIRED_QT_VERSION} CONFIG REQUIRED Core Widgets)
-
-include(KDEInstallDirs)
-include(KDEFrameworkCompilerSettings NO_POLICY_SCOPE)
-include(KDECMakeSettings)
-include(FeatureSummary)
-
-
-ADD_SUBDIRECTORY(src)
-ADD_SUBDIRECTORY(doc)
-
-#if("${CMAKE_BINARY_DIR}" STREQUAL "{CMAKE_CURRENT_BINARY_DIR}")
-    feature_summary(WHAT ALL FATAL_ON_MISSING_REQUIRED_PACKAGES)
-#endif()
+add_subdirectory(src)
 
+if (KF5DocTools_FOUND)
+    add_subdirectory(doc)
+endif()
