diff --git a/.qmake.conf b/.qmake.conf
index 9d4004a0..b07c1dec 100644
--- a/.qmake.conf
+++ b/.qmake.conf
@@ -3,3 +3,18 @@ load(qt_build_config)
 CONFIG += warning_clean
 
 MODULE_VERSION = 5.8.0
+
+QMAKE_CFLAGS -= -g
+QMAKE_CFLAGS -= -gstabs
+QMAKE_CXXFLAGS -= -g
+QMAKE_CXXFLAGS -= -gstabs
+QT_CONFIG -= force_debug_info
+QT_CONFIG -= separate_debug_info
+
+QMAKE_CXXFLAGS += --param ggc-min-expand=20
+QMAKE_CFLAGS += --param ggc-min-expand=20
+QMAKE_LFLAGS -= Wl,--gc-sections
+QMAKE_LFLAGS -= -fuse-ld=gold
+QMAKE_LFLAGS += -fuse-ld=bfd
+
+WEBENGINE_CONFIG+=reduce_binary_size
